src/main/java/macrobase/analysis/pipeline/BasicContextualBatchedPipeline.java

```diff
@@ -0,0 +1,128 @@
+package macrobase.analysis.pipeline;
+
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.collect.Lists;
+
+import macrobase.analysis.classify.OutlierClassifier;
+import macrobase.analysis.contextualoutlier.Context;
+import macrobase.analysis.contextualoutlier.ContextualOutlierDetector;
+import macrobase.analysis.result.AnalysisResult;
+import macrobase.analysis.result.OutlierClassificationResult;
+import macrobase.analysis.summary.BatchSummarizer;
+import macrobase.analysis.summary.Summary;
+import macrobase.conf.ConfigurationException;
+import macrobase.conf.MacroBaseConf;
+import macrobase.datamodel.Datum;
+import macrobase.ingest.DataIngester;
+import macrobase.util.IterUtils;
+
+public class BasicContextualBatchedPipeline extends AbstractPipeline {
```

Comment:
Since BasicContextualBatchedPipeline produces multiple contexts that have outliers, it is convenient to use the iterator in AbstractPipeline to return next contextual outlier by calling next() and hasNext().